### PR TITLE
[enh] Revive the old auto documentation of API with swagger

### PR DESCRIPTION
**Pull request by zamentur**
## The problem

...

## Solution

...

## PR Status

...

## How to test

...

### CODE REVIEW

File: doc/bash_completion.d/yunohost [hunk 1]
```diff
@@ -0,0 +1,125 @@
+#
+# completion for yunohost
+# automatically generated from the actionsmap
+#
```
Comment (alexAubin) [inline doc/bash_completion.d/yunohost:4]: This file is not meant to be commited é_è

File: doc/openapi.json [hunk 2]
```diff
@@ -0,0 +1 @@
...o be ignored and therefore not considered as actual issues", "description": "", "responses": {"200": {"description": "successful operation"}}, "parameters": [], "requestBody": {"required": true, "content": {"multipart/form-data": {"schema": {"type": "object", "properties": {"filter": {"type": "array", "description": "Add a filter. The first element should be a diagnosis category, and other criterias can be provided using the infos from the 'meta' sections in 'yunohost diagnosis show'. For example: 'dnsrecords domain=yolo.test category=xmpp'"}, "list": {"type": "boolean", "description": "List active ignore filters"}}, "required": []}}}}}}, "/diagnosis/unignore": {"put": {"tags": ["diagnosis"], "operationId": "diagnosis_unignore", "summary": "Configure some diagnosis results to be unignored and therefore considered as actual issues", "description": "", "responses": {"200": {"description": "successful operation"}}, "parameters": [], "requestBody": {"required": true, "content": {"multipart/form-data": {"schema": {"type": "object", "properties": {"filter": {"type": "array", "description": "Remove a filter (it should be an existing filter as listed with --list)"}}, "required": []}}}}}}}}
```
Comment (alexAubin) [inline doc/openapi.json:1]: Sounds like this file is not supposed to be commited either ?

Comment (alexAubin) [inline doc/openapi.json:1]: Sounds like this file is not supposed to be commited either ?

File: doc/openapi.js [hunk 3]
```diff
@@ -0,0 +1 @@
...o be ignored and therefore not considered as actual issues", "description": "", "responses": {"200": {"description": "successful operation"}}, "parameters": [], "requestBody": {"required": true, "content": {"multipart/form-data": {"schema": {"type": "object", "properties": {"filter": {"type": "array", "description": "Add a filter. The first element should be a diagnosis category, and other criterias can be provided using the infos from the 'meta' sections in 'yunohost diagnosis show'. For example: 'dnsrecords domain=yolo.test category=xmpp'"}, "list": {"type": "boolean", "description": "List active ignore filters"}}, "required": []}}}}}}, "/diagnosis/unignore": {"put": {"tags": ["diagnosis"], "operationId": "diagnosis_unignore", "summary": "Configure some diagnosis results to be unignored and therefore considered as actual issues", "description": "", "responses": {"200": {"description": "successful operation"}}, "parameters": [], "requestBody": {"required": true, "content": {"multipart/form-data": {"schema": {"type": "object", "properties": {"filter": {"type": "array", "description": "Remove a filter (it should be an existing filter as listed with --list)"}}, "required": []}}}}}}}}
```
Comment (alexAubin) [inline doc/openapi.js:1]: Same here ?

Comment (zamentur) [reply doc/openapi.js:1]: i think we should add it here to ease the way to keep our swagger api doc up to date

File: doc/bash_completion.d/yunohost (hunk 1 above)
Comment (zamentur) [reply doc/bash_completion.d/yunohost:4]: should we add it inside .gitignore ?

File: doc/openapi.js (hunk 3 above)
Comment (alexAubin) [reply doc/openapi.js:1]: I'm not sure to understand how it helps to keep it up to date ... Is the API doc meant to be hosted in the yunohost source ? Or is it meant to be hosted in the Grav doc ? Or both ? If it's meant to be hosted in the yunohost source, what's the process to automatically keep it up to date ? Shall we add a job in the gitlab CI conf ?

File: doc/bash_completion.d/yunohost (hunk 1 above)
Comment (alexAubin) [reply doc/bash_completion.d/yunohost:4]: sure

**Comment by alexAubin** [comment]
Merging, added an `./ynh-dev rebuild-api-doc` to ynh-dev to autoregen the swagger files

**Pull request merged**